# Below are the changes needed for generate-gn-build-files.sh
# except for depot_tools
# https://bugzilla.mozilla.org/show_bug.cgi?id=1654448

new file mode 100644
--- dom/media/webrtc/third_party_build/gn-configs/arm64_False_arm64_freebsd.mozconfig.orig	2022-11-06 12:36:18 UTC
+++ dom/media/webrtc/third_party_build/gn-configs/arm64_False_arm64_freebsd.mozconfig
@@ -0,0 +1,12 @@
+export M4=/usr/local/bin/gm4
+export CC=/usr/local/bin/clang13
+export CXX=/usr/local/bin/clang++13
+export CPP=/usr/local/bin/clang-cpp13
+ac_add_options --prefix="/usr/local"
+ac_add_options --with-libclang-path="/usr/local/llvm13/lib"
+ac_add_options --with-wasi-sysroot="/usr/local/share/wasi-sysroot"
+
+ac_add_options --target=aarch64
+ac_add_options --enable-bootstrap
+
+mk_add_options MOZ_OBJDIR=obj-arm64_False_arm64_freebsd
--- dom/media/webrtc/third_party_build/gn-configs/arm64_True_arm64_freebsd.mozconfig.orig	2022-11-06 12:36:18 UTC
+++ dom/media/webrtc/third_party_build/gn-configs/arm64_True_arm64_freebsd.mozconfig
@@ -0,0 +1,13 @@
+export M4=/usr/local/bin/gm4
+export CC=/usr/local/bin/clang13
+export CXX=/usr/local/bin/clang++13
+export CPP=/usr/local/bin/clang-cpp13
+ac_add_options --prefix="/usr/local"
+ac_add_options --with-libclang-path="/usr/local/llvm13/lib"
+ac_add_options --with-wasi-sysroot="/usr/local/share/wasi-sysroot"
+
+ac_add_options --target=aarch64
+ac_add_options --enable-debug
+ac_add_options --enable-bootstrap
+
+mk_add_options MOZ_OBJDIR=obj-arm64_True_arm64_freebsd
--- dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_freebsd.mozconfig.orig	2022-11-06 12:42:21 UTC
+++ dom/media/webrtc/third_party_build/gn-configs/x64_False_x64_freebsd.mozconfig
@@ -0,0 +1,11 @@
+export M4=/usr/local/bin/gm4
+export CC=/usr/local/bin/clang13
+export CXX=/usr/local/bin/clang++13
+export CPP=/usr/local/bin/clang-cpp13
+ac_add_options --prefix="/usr/local"
+ac_add_options --with-libclang-path="/usr/local/llvm13/lib"
+ac_add_options --with-wasi-sysroot="/usr/local/share/wasi-sysroot"
+
+ac_add_options --enable-bootstrap
+
+mk_add_options MOZ_OBJDIR=obj-x64_False_x64_freebsd
--- dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_freebsd.mozconfig.orig	2022-11-06 12:42:21 UTC
+++ dom/media/webrtc/third_party_build/gn-configs/x64_False_x86_freebsd.mozconfig
@@ -0,0 +1,12 @@
+export M4=/usr/local/bin/gm4
+export CC=/usr/local/bin/clang13
+export CXX=/usr/local/bin/clang++13
+export CPP=/usr/local/bin/clang-cpp13
+ac_add_options --prefix="/usr/local"
+ac_add_options --with-libclang-path="/usr/local/llvm13/lib"
+ac_add_options --with-wasi-sysroot="/usr/local/share/wasi-sysroot"
+
+ac_add_options --target=i686
+ac_add_options --enable-bootstrap
+
+mk_add_options MOZ_OBJDIR=obj-x64_False_x86_freebsd
--- dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_freebsd.mozconfig.orig	2022-11-06 12:42:21 UTC
+++ dom/media/webrtc/third_party_build/gn-configs/x64_True_x64_freebsd.mozconfig
@@ -0,0 +1,12 @@
+export M4=/usr/local/bin/gm4
+export CC=/usr/local/bin/clang13
+export CXX=/usr/local/bin/clang++13
+export CPP=/usr/local/bin/clang-cpp13
+ac_add_options --prefix="/usr/local"
+ac_add_options --with-libclang-path="/usr/local/llvm13/lib"
+ac_add_options --with-wasi-sysroot="/usr/local/share/wasi-sysroot"
+
+ac_add_options --enable-debug
+ac_add_options --enable-bootstrap
+
+mk_add_options MOZ_OBJDIR=obj-x64_True_x64_freebsd
--- dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_freebsd.mozconfig.orig	2022-11-06 12:42:21 UTC
+++ dom/media/webrtc/third_party_build/gn-configs/x64_True_x86_freebsd.mozconfig
@@ -0,0 +1,13 @@
+export M4=/usr/local/bin/gm4
+export CC=/usr/local/bin/clang13
+export CXX=/usr/local/bin/clang++13
+export CPP=/usr/local/bin/clang-cpp13
+ac_add_options --prefix="/usr/local"
+ac_add_options --with-libclang-path="/usr/local/llvm13/lib"
+ac_add_options --with-wasi-sysroot="/usr/local/share/wasi-sysroot"
+
+ac_add_options --target=i686
+ac_add_options --enable-debug
+ac_add_options --enable-bootstrap
+
+mk_add_options MOZ_OBJDIR=obj-x64_True_x86_freebsd
--- third_party/libwebrtc/build/config/BUILDCONFIG.gn.orig	2022-11-04 13:53:20 UTC
+++ third_party/libwebrtc/build/config/BUILDCONFIG.gn
@@ -184,7 +184,7 @@ if (host_toolchain == "") {
   # TODO(dpranke): Add some sort of assert here that verifies that
   # no toolchain omitted host_toolchain from its toolchain_args().
 
-  if (host_os == "linux" || host_os == "openbsd") {
+  if (host_os == "linux" || host_os == "openbsd" || host_os == "freebsd" ) {
     if (target_os != "linux") {
       host_toolchain = "//build/toolchain/linux:clang_$host_cpu"
     } else if (is_clang) {
@@ -222,7 +222,8 @@ if (target_os == "android") {
   assert(host_os == "linux" || host_os == "mac",
          "Android builds are only supported on Linux and Mac hosts.")
   _default_toolchain = "//build/toolchain/android:android_clang_$target_cpu"
-} else if (target_os == "chromeos" || target_os == "linux" || target_os == "openbsd") {
+} else if (target_os == "chromeos" || target_os == "linux" ||
+           target_os == "openbsd" || target_os == "freebsd") {
   # See comments in build/toolchain/cros/BUILD.gn about board compiles.
   if (is_clang) {
     _default_toolchain = "//build/toolchain/linux:clang_$target_cpu"
